compiler/cmdlinehelper.nim

```diff
@@ -69,8 +69,9 @@ proc loadConfigsAndRunMainCommand*(self: NimProg, cache: IdentCache; conf: Confi
     let scriptFile = conf.projectFull.changeFileExt("nims")
     if not self.suggestMode:
       runNimScriptIfExists(scriptFile)
-      # 'nim foo.nims' means to just run the NimScript file and do nothing more:
-      if fileExists(scriptFile) and scriptFile == conf.projectFull:
```

Comment:
@Wh1teDuke  indeed what @araq said; otherwise there's a bug ; to make this more precise, how about something like:
```nim
if fileExists(scriptFile):
    if scriptFile == conf.projectFull:
      doAssert conf.command in ["nop", "e"] # or use error, etc, but with same semantics
      return false
else:
  # rest of existing code
```

Reply:
I'll find another way, but that check is what triggers the error in the issue (`nim test script t1 t2 t3` would evaluate `scriptFile == conf.projectFull` as false since `test.nims != test`)